packages/http/src/types.ts

```diff
@@ -500,6 +510,53 @@ export interface HttpOperationPart {
   readonly multi: boolean;
 }
 
+/**
+ * The type of an HTTP body that is a file upload or download.
+ */
+export interface HttpOperationFileBody extends HttpOperationBodyBase {
+  readonly bodyKind: "file";
+  /**
+   * The model type of the body that is or extends `Http.File`.
+   */
+  readonly type: Model;
```

Comment:
Is this the http namesapce file? If I extend this, would it be `File` or `SpecFile`?

```ts
model SpecFile extends File {
  // ...
}
```

Reply:
It's the type that was recognized as an instance of File. So in this example it would be a reference to SpecFile, not Http.File. Or in the spread case like `op foo(...File): void;` it would be a direct reference to the parameters of `foo`.